packages/paste-core/layout/flex/__tests__/flex.test.tsx

```diff
@@ -198,261 +199,280 @@ describe('Flex Unit Tests', () => {
-            </Flex>
-            <Flex>
-              <Box backgroundColor="colorBackgroundBrandHighlight" minWidth="size20" minHeight="size10" />
-            </Flex>
-          </Flex>
-        </Theme.Provider>
-      )
-      .toJSON();
-    expect(tree).toMatchSnapshot();
+    render(
+      <Theme.Provider theme="console">
+        <Flex data-testid="renderedFlexElement" hAlignContent={['left', 'center', 'right']} />
+      </Theme.Provider>
+    );
+    const renderedFlex = screen.getByTestId('renderedFlexElement');
+    expect(renderedFlex).toHaveStyleRule('justify-content', 'center', {
+      media: 'screen and (min-width:25rem)',
+    });
+    expect(renderedFlex).toHaveStyleRule('justify-content', 'flex-end', {
+      media: 'screen and (min-width:64rem)',
+    });
+    expect(renderedFlex).toHaveStyleRule('justify-content', 'flex-start');
+  });
+});
+describe('Flex style props', () => {
+  it('should set all available pass through style props', (): void => {
+    render(
+      <Flex
+        data-testid="renderedFlexElement"
+        margin="space20"
+        padding="space20"
+        maxWidth="space20"
+        minWidth="space20"
+        minHeight="space20"
+        maxHeight="space20"
+      />
+    );
+    const renderedFlex = screen.getByTestId('renderedFlexElement');
+    expect(renderedFlex).toHaveStyleRule('margin', 'space20');
+    expect(renderedFlex).toHaveStyleRule('padding', 'space20');
+    expect(renderedFlex).toHaveStyleRule('max-width', 'space20');
+    expect(renderedFlex).toHaveStyleRule('min-width', 'space20');
+    expect(renderedFlex).toHaveStyleRule('min-height', 'space20');
+    expect(renderedFlex).toHaveStyleRule('max-height', 'space20');
+  });
+
+  it('should set individual width and height style props', (): void => {
+    render(<Flex data-testid="renderedFlexElement" width="size20" height="size20" />);
+    const renderedFlex = screen.getByTestId('renderedFlexElement');
+    expect(renderedFlex).toHaveStyleRule('width', 'size20');
+    expect(renderedFlex).toHaveStyleRule('height', 'size20');
+  });
+
+  it('should set individual margin style props', (): void => {
+    render(
+      <Flex
+        data-testid="renderedFlexElement"
+        marginTop="space20"
+        marginRight="space20"
+        marginBottom="space20"
+        marginLeft="space20"
+      />
+    );
+    const renderedFlex = screen.getByTestId('renderedFlexElement');
+    expect(renderedFlex).toHaveStyleRule('margin-top', 'space20');
+    expect(renderedFlex).toHaveStyleRule('margin-right', 'space20');
+    expect(renderedFlex).toHaveStyleRule('margin-bottom', 'space20');
+    expect(renderedFlex).toHaveStyleRule('margin-left', 'space20');
+  });
+
+  it('should set individual padding style props', (): void => {
+    render(
+      <Flex
+        data-testid="renderedFlexElement"
+        paddingTop="space20"
+        paddingRight="space20"
+        paddingBottom="space20"
+        paddingLeft="space20"
```

Comment:
Looking at these tests is really useful for thinking of the ones I'm writing right now (seeing the modularity especially) 🤓
Would it make sense to have these attributes be all different space options to make absolutely certain that the styles match up exactly, or is that overkill?

Reply:
Maybe? It certainly doesn't hurt, so I'll tweak them